### ISSUE TITLE
Can't select first option from Picker

### ISSUE THREAD

**Issue by mayconmesquita** [facebook/react-native#28204]
## Overview
Using a Picker, onValueChange has no effect when the first item is selected, but works for the rest.

When pressing 1st item, for the first time, this won't fire, because is selected by RN by default (even if my state value is undefined/false/0/'whatever').

## React Native version:

System:
    OS: Windows 10 10.0.18362
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 6.03 GB / 15.85 GB
  Binaries:
    Node: 11.10.1 - C:\Program Files\nodejs\node.EXE
    npm: 6.7.0 - C:\Program Files\nodejs\npm.CMD
  SDKs:
    Android SDK:
      API Levels: 23, 28, 29
      Build Tools: 28.0.3, 29.0.2
      System Images: android-28 | Google Play Intel x86 Atom, android-29 | Google APIs Intel x86 Atom
  IDEs:
    Android Studio: Version  3.5.0.0 AI-191.8026.42.35.6010548
  npmPackages:
    react: 16.12.0 => 16.12.0
    react-native: 0.61.5 => 0.61.5

## Steps To Reproduce

I created this **Snack** showing the issue. All you need to do is **Tap to Play**.

https://snack.expo.io/@mayconjcm/picker-rn-bug

## Expected Results

We need to have some function or callback, to know when the first picker item is selected. My workaround is creating a last picker item: `<Picker.Item label="Cancel" value="none" />`. But this solution is not most appropriate.

## Snack, code example, screenshot, or link to a repository:

https://snack.expo.io/@mayconjcm/picker-rn-bug

**Comment by Djunnni** [comment]
~~~ 
  state = {
    value: undefined,
  };

<Picker
          selectedValue={this.state.value}
          style={{ flex: 1 }}
          onValueChange={this.changeOption}
          mode="dropdown">
          <Picker.Item label="Options"/>
          <Picker.Item label="Refresh" value="refresh" />
          <Picker.Item label="Share" value="share" />
          <Picker.Item label="Go To Broswer" value="openExternal" />
</Picker>
~~~
As you know, Picker selects first item by Default.
i recommend to use like 'Options' that don't have value.

**Comment by mayconmesquita** [comment]
My workaround is creating a last picker item: <Picker.Item label="Cancel" value="none" />.
But this solution is not most appropriate, because is a useless option and helps to get confused.

I would like to get a result like Gmail or Chrome, options picker (see my gif).

![ezgif-2-76370fbc346c](https://user-images.githubusercontent.com/46308804/75698993-70744180-5c8e-11ea-8dfb-d00891b24637.gif)

**Comment by nxr5013** [comment]
> ```
>   state = {
>     value: undefined,
>   };
> 
> <Picker
>           selectedValue={this.state.value}
>           style={{ flex: 1 }}
>           onValueChange={this.changeOption}
>           mode="dropdown">
>           <Picker.Item label="Options"/>
>           <Picker.Item label="Refresh" value="refresh" />
>           <Picker.Item label="Share" value="share" />
>           <Picker.Item label="Go To Broswer" value="openExternal" />
> </Picker>
> ```
> 
> As you know, Picker selects first item by Default.
> i recommend to use like 'Options' that don't have value.
> 
> if you need some of help, question to StackOverflow :)

The way you have implemented the Picker, from a conceptual level, is completely correct. The reason you are getting undefined is because you are literally passing in that. The picker will print and show exactly what you want as long as you pass in the required values. I have recoded your small example and made it dynamic that would allow you to make improvements and modification easier. 

https://snack.expo.io/S1_ipbwSL

**Comment by mayconmesquita** [comment]
@nxr5013 Thanks for the code refactoring. 

I think the correct/ideal behavior, is **onValueChange** fires, even if **selectedValue** is undefined or empty value. So then, we would have a way for init the picker unselected.

So with the fix, we could have a picker acting as a **options menu** with all options working and firing when you press.

You can see this 'widget' working in Chrome/Whatsapp Web/Gmail's App/Most apps.

**Comment by LuizDoPc** [comment]
could we get a patch to 0.61.2 fixing this? Or we would have to update RN version?

**Comment by mayconmesquita** [comment]
@LuizDoPc We don't have a fix implemented yet.

**Comment by dushyanttessact** [comment]
```
<Picker
 selectedValue={this.state.selected_tag ? this.state.selected_tag.title : "---Select---"}
 style={{height: 40, width: 100, color: 'transparent', fontSize: 20}}
 itemStyle={{fontSize: 14, color: '#dbdfe2', zIndex: 100}}
 mode="dropdown"
 onValueChange={this.setSelected_tag}>
 {this.props.compliance_reducer.tag_list && this.props.compliance_reducer.tag_list.map((item, i)=>{
   return <Picker.Item key={i} label={item.title} value={item} />
  })}
</Picker>
```

I can't pick first value from this, is there any solution yet?

**Comment by LuizDoPc** [comment]
@dushyanttessact i'm using this workaround 

> ```
>   state = {
>     value: undefined,
>   };
> 
> <Picker
>           selectedValue={this.state.value}
>           style={{ flex: 1 }}
>           onValueChange={this.changeOption}
>           mode="dropdown">
>           <Picker.Item label="Options"/>
>           <Picker.Item label="Refresh" value="refresh" />
>           <Picker.Item label="Share" value="share" />
>           <Picker.Item label="Go To Broswer" value="openExternal" />
> </Picker>
> ```
> 
> As you know, Picker selects first item by Default.
> i recommend to use like 'Options' that don't have value.
> 
> if you need some of help, question to StackOverflow :)

**Comment by mayconmesquita** [comment]
We don't have a solution yet.

**Comment by jackel119** [comment]
Same problem. `onValueChanged` doesn't fire if the first item is picked.

**Comment by mayconmesquita** [comment]
@jackel119 Yes. I think the easiest way to get this is working today is, patching the java directly and adding to the project using `patch-package`.

**Comment by jackel119** [comment]
@mayconmesquita Can you share your native fix if you already have one?
